lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -420,7 +421,10 @@ def template_exists?(template_endpoint, name)
     def template_put(template_endpoint, name, template)
       path = "#{template_endpoint}/#{name}"
       logger.info("Installing Elasticsearch template", name: name)
-      @pool.put(path, nil, LogStash::Json.dump(template))
+      response = @pool.put(path, nil, LogStash::Json.dump(template))
+    rescue ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
+      return response if e.code == 404
+      raise e
```

Comment:
we're not doing anything with the return value, and we can't guarantee `response` will be defined in the rescue block:

```suggestion
      @pool.put(path, nil, LogStash::Json.dump(template))
    rescue ::LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
      raise e if e.code != 404
```

Reply:
Totally, i had misunderstood something earlier and added that by mistake. Updated in [93eddbc](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1199/commits/93eddbc420445b11fba2d10906a9e5c162e40f5b)